File: backend/strip_frontend_comments.js
const fs = require('fs');
const path = require('path');

function removeComments(filePath) {
    try {
        let content = fs.readFileSync(filePath, 'utf8');
        // Remove multi-line comments
        content = content.replace(/\/\*[\s\S]*?\*\//g, '');
        // Remove single-line comments (careful not to match http://)
        content = content.replace(/(?<!:)\/\/.*/g, '');
        
        fs.writeFileSync(filePath, content, 'utf8');
        console.log(`Stripped: ${filePath}`);
    } catch (e) {
        console.error(`Skipped ${filePath}: ${e.message}`);
    }
}

function processDirectory(dir) {
    if (!fs.existsSync(dir)) return;
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            if (file !== 'node_modules' && file !== '.next' && file !== 'dist' && !file.startsWith('.')) {
                processDirectory(fullPath);
            }
        } else if (file.match(/\.(js|jsx|ts|tsx)$/)) {
            removeComments(fullPath);
        }
    }
}

processDirectory('./frontend');

File: backend/strip_frontend_comments.py
import os
import re

def remove_comments(file_path):
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        content = re.sub(r'/\*[\s\S]*?\*/', '', content)
    
        content = re.sub(r'(?<!:)//.*', '', content)
        
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Stripped: {file_path}")
    except Exception as e:
        print(f"Skipped {file_path}: {e}")

if os.path.exists('frontend'):
    for root, dirs, files in os.walk('frontend'):
        if 'node_modules' in dirs:
            dirs.remove('node_modules')
        if '.next' in dirs:
            dirs.remove('.next')
        for file in files:
            if file.endswith(('.js', '.jsx', '.ts', '.tsx', '.css')):
                remove_comments(os.path.join(root, file))
